test-data/run.test

```diff
@@ -375,11 +375,106 @@ assert 3 in val
 assert 5 in val
 assert 8 in val
 assert len(val) == 5
+assert val == {1, 2, 3, 5, 8}
 s = 0
 for i in val:
     s += i
 assert s == 19
 
+[case testSetFromIterable]
+from typing import Set, List
+def f() -> List[Set[int]]:
+    x = set([1, 3, 5])
+    y = set((1, 3, 5))
+    z = set({1: '1', 3: '3', 5: '5'})
+    return [x, y, z]
+[file driver.py]
+from native import f
+sets = f()
+for s in sets:
+    assert s == {1, 3, 5}
+
+[case testSetAdd]
+from typing import Set
+def f(s : Set[int]) -> None:
+    for a in [1, 2, 3]:
+        if a not in s:
+            s.add(a)
+            return
+[file driver.py]
+from native import f
+s = set()
+f(s)
+assert s == {1}
+f(s)
+assert s == {1, 2}
+f(s)
+assert s == {1, 2, 3}
+
+[case testSetClear]
+from typing import Set
+def f(s : Set[int]) -> None:
+    s.clear()
+    s.add(1)
+[file driver.py]
+from native import f
+s = {3, 7, 12}
+f(s)
+assert s == {1}
+
+[case testSetRemove]
+from typing import Set
+def f(s: Set[int]) -> None:
+    s.remove(1)
+[file driver.py]
+from native import f
+import traceback
+s = {1, 4, 6}
+f(s)
+assert s == {4, 6}
+try:
+    f(s)
+except KeyError as e:
+    print(e)
+[out]
+1
+
+[case testSetDiscard]
+from typing import Set
+def f(s: Set[int]) -> None:
+    s.discard(1)
+[file driver.py]
+from native import f
+import traceback
+s = {1, 4, 6}
+f(s)
+assert s == {4, 6}
+f(s)
+assert s == {4, 6}
```

Comment:
This is going to seem tasteless, because it is, but:
Could you merge all of the run tests for sets into one test? There is a lot of per-test overhead for run tests.